i18n-helpers/src/bin/markdown-gettext.rs

```diff
@@ -0,0 +1,248 @@
+
+fn translate_files(
+    catalog: Catalog,
+    files: Vec<PathBuf>,
+    output_path: PathBuf,
+) -> anyhow::Result<()> {
+    for file_path in files.iter() {
+        let content = File::open(&file_path)
+            .with_context(|| format!("Failed to open file: {}", file_path.display()))?;
+        let mut buf_reader = BufReader::new(content);
+        let mut contents = String::new();
+        buf_reader
+            .read_to_string(&mut contents)
+            .with_context(|| format!("Failed to read file: {}", file_path.display()))?;
+
+        let translated_content = translate(&contents, &catalog);
+
+        let output_file_name = file_path.file_name().ok_or(anyhow!("Invalid file name"))?;
+        let output_file_path = output_path.join(output_file_name);
+
+        let mut output_file = File::create(&output_file_path)
+            .with_context(|| format!("Failed to create file: {}", output_file_path.display()))?;
+
+        output_file
+            .write_all(translated_content.as_bytes())
+            .with_context(|| format!("Failed to write to file: {}", output_file_path.display()))?;
+    }
+
+    Ok(())
+}
+
+fn validate_file(path: &PathBuf) -> bool {
+    path.is_file() && path.extension().map_or(false, |ext| ext == "md")
+}
+
+fn allocate_files(
+    dir_option: Option<String>,
+    file_option: Option<String>,
+) -> anyhow::Result<Vec<PathBuf>> {
+    let mut valid_files = Vec::new();
+
+    match (dir_option, file_option) {
+        (Some(_), Some(_)) => return bail!("Only one of --dir or --f should be specified"),
+        (None, None) => return bail!("Either --dir or --f must be specified"),
+        (Some(dir_path), None) => {
+            let full_dir = PathBuf::from(&dir_path);
+            if !full_dir.exists() || !full_dir.is_dir() {
+                return bail!("Directory does not exist: {}", full_dir.display());
+            }
+
+            for entry in fs::read_dir(full_dir)? {
+                let entry = entry?;
+                let path = entry.path();
+
+                if validate_file(&path) {
+                    valid_files.push(path);
+                }
+            }
+        }
+        (None, Some(file_path)) => {
+            let full_file_path = PathBuf::from(&file_path);
+            if !validate_file(&full_file_path) {
+                bail!("Markdown file does not exist: {}", full_file_path.display())
+            }
+            valid_files.push(full_file_path);
+        }
+        _ => unreachable!(),
+    }
+
+    Ok(valid_files)
+}
+
+fn find_output_path(output_path_option: Option<String>) -> anyhow::Result<PathBuf> {
+    match output_path_option {
+        Some(output_path) => {
+            let path = PathBuf::from(&output_path);
+            if path.exists() {
+                if !path.is_dir() {
+                    bail!(
+                        "Specified output path exists but is not a directory: {}",
+                        path.display()
+                    );
+                }
```

Comment:
Like above, you should skip the error handling here and let `fs::create_dir_all` below raise an error if there is a problem.